core/templates/dev/head/pages/exploration_player/TutorCardDirective.js

```diff
@@ -0,0 +1,165 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for the Tutor Card.
+ */
+
+oppia.animation('.conversation-skin-responses-animate-slide', function() {
+  return {
+    removeClass: function(element, className, done) {
+      if (className !== 'ng-hide') {
+        done();
+        return;
+      }
+      element.hide().slideDown(400, done);
+    },
+    addClass: function(element, className, done) {
+      if (className !== 'ng-hide') {
+        done();
+        return;
+      }
+      element.slideUp(400, done);
+    }
+  };
+});
+
+oppia.animation('.conversation-skin-animate-tutor-card-content', function() {
+  var animateCardChange = function(element, className, done) {
+    if (className !== 'animate-card-change') {
+      return;
+    }
+
+    var currentHeight = element.height();
+    var expectedNextHeight = $(
+      '.conversation-skin-future-tutor-card ' +
+      '.conversation-skin-tutor-card-content'
+    ).height();
+
+    // Fix the current card height, so that it does not change during the
+    // animation, even though its contents might.
+    element.css('height', currentHeight);
+
+    jQuery(element).animate({
+      opacity: 0
+    }, TIME_FADEOUT_MSEC).animate({
+      height: expectedNextHeight
+    }, TIME_HEIGHT_CHANGE_MSEC).animate({
+      opacity: 1
+    }, TIME_FADEIN_MSEC, function() {
+      element.css('height', '');
+      done();
+    });
+
+    return function(cancel) {
+      if (cancel) {
+        element.css('opacity', '1.0');
+        element.css('height', '');
+        element.stop();
+      }
+    };
+  };
+
+  return {
+    addClass: animateCardChange
+  };
+});
+
+oppia.directive('tutorCard', [function() {
+  return {
+    restrict: 'E',
+    scope: {
+      startCardChangeAnimation: '=',
+      showNextCard: '&',
+      showSupplementalCard: '&',
+      onSubmitAnswer: '&'
+    },
+    templateUrl: 'components/TutorCard',
+    controller: [
+    '$scope', 'oppiaPlayerService', 'UrlInterpolationService',
+    'playerPositionService', 'playerTranscriptService',
+    'explorationStateService', 'windowDimensionsService',
+    function($scope, oppiaPlayerService, UrlInterpolationService,
+      playerPositionService, playerTranscriptService, explorationStateService,
```

Comment:
nit: suggest moving explorationStateService to next line for parity with the strings in the array.